packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -12,12 +12,187 @@ export interface CodeBuildStartBuildProps extends sfn.TaskStateBaseProps {
    * CodeBuild project to start
    */
   readonly project: codebuild.IProject;
+  /**
+   * Build output artifact settings that override, for this build only.
+   *
+   * @default - the latest ones already defined in the build project.
+   */
+  readonly artifactsOverride?: codebuild.IArtifacts;
+  /**
+   * A buildspec file declaration that overrides, for this build only.
+   *
+   * @default - the latest one already defined in the build project.
+   */
+  readonly buildspecOverride?: string;
+  /**
+   * Contains information that defines how the build project reports the build status to the source provider.
+   *
+   * @default - the latest one already defined in the build project only if the build's source is GitHub, GitHub Enterprise, or Bitbucket.
+   */
+  readonly buildStatusConfigOverride?: codebuild.CfnProject.BuildStatusConfigProperty;
+  /**
+   * A ProjectCache object specified for this build.
+   *
+   * @default - the one defined in the build project.
+   */
+  readonly cacheOverride?: codebuild.CfnProject.ProjectCacheProperty;
+  /**
+   * The name of a certificate for this build.
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly certificateOverride?: string;
+  /**
+   * The name of a compute type for this build
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly computeTypeOverride?: codebuild.ComputeType;
```

Comment:
this is one of the properties in `codebuild.BuildEnvironment` should that just be accepted as an override parameter and it's values can be used accordingly?